relay_messages.go

```diff
@@ -96,8 +96,12 @@ type lazyCallReq struct {
 
 	caller, method, delegate, key, as []byte
 
-	arg2StartOffset, arg2EndOffset int
-	isArg2Fragmented               bool
+	checksumTypeOffset             uint16
```

Comment:
good idea using `uint16` for the offsets, saves some space!

you might want to play with struct alignment too (although I think what you have looks good already)